Extension/src/LanguageServer/extension.ts

```diff
@@ -499,110 +497,99 @@ function installVsix(vsixLocation: string): Thenable<void> {
 
     // Get the path to the VSCode command -- replace logic later when VSCode allows calling of
     // workbench.extensions.action.installVSIX from TypeScript w/o instead popping up a file dialog
-    return PlatformInformation.GetPlatformInformation().then((platformInfo) => {
-        const vsCodeScriptPath: string | undefined = function(platformInfo): string | undefined {
-            if (platformInfo.platform === 'win32') {
-                const vsCodeBinName: string = path.basename(process.execPath);
-                let cmdFile: string; // Windows VS Code Insiders/Exploration breaks VS Code naming conventions
-                if (vsCodeBinName === 'Code - Insiders.exe') {
-                    cmdFile = 'code-insiders.cmd';
-                } else if (vsCodeBinName === 'Code - Exploration.exe') {
-                    cmdFile = 'code-exploration.cmd';
-                } else {
-                    cmdFile = 'code.cmd';
-                }
-                const vsCodeExeDir: string = path.dirname(process.execPath);
-                return path.join(vsCodeExeDir, 'bin', cmdFile);
-            } else if (platformInfo.platform === 'darwin') {
-                return path.join(process.execPath, '..', '..', '..', '..', '..',
-                    'Resources', 'app', 'bin', 'code');
+    const platformInfo: PlatformInformation = await PlatformInformation.GetPlatformInformation();
+    const getVsCodeScriptPath = (platformInfo: any): string => {
+        if (platformInfo.platform === 'win32') {
+            const vsCodeBinName: string = path.basename(process.execPath);
+            let cmdFile: string; // Windows VS Code Insiders/Exploration breaks VS Code naming conventions
+            if (vsCodeBinName === 'Code - Insiders.exe') {
+                cmdFile = 'code-insiders.cmd';
+            } else if (vsCodeBinName === 'Code - Exploration.exe') {
+                cmdFile = 'code-exploration.cmd';
             } else {
-                const vsCodeBinName: string = path.basename(process.execPath);
-                try {
-                    return which.sync(vsCodeBinName);
-                } catch (error) {
-                    return undefined;
-                }
+                cmdFile = 'code.cmd';
             }
-        }(platformInfo);
-        if (!vsCodeScriptPath) {
-            return Promise.reject(new Error('Failed to find VS Code script'));
-        }
-
-        // 1.28.0 changes the CLI for making installations.  1.27.2 was immediately prior.
-        const oldVersion: PackageVersion = new PackageVersion('1.27.2');
-        if (userVersion.isVsCodeVersionGreaterThan(oldVersion)) {
-            return new Promise<void>((resolve, reject) => {
```

Comment:
I believe this needs to stay as a Promise.  Those `reject` statements that were changed to `throw` are not running in this context and as a result cannot be caught.